File: src/main/java/com/gmail/borlandlp/minigamesdtools/activepoints/reaction/Reaction.java
package com.gmail.borlandlp.minigamesdtools.activepoints.reaction;

import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint;
import org.bukkit.entity.Entity;
import org.bukkit.entity.LivingEntity;

public abstract class Reaction {
    protected ActivePoint activePoint;

    public void setActivePoint(ActivePoint activePoint) {
        this.activePoint = activePoint;
    }

    public ActivePoint getActivePoint() {
        return activePoint;
    }

    public abstract void performDamage(Entity reactionInitiator, double damage);
    public abstract void performIntersection(Entity reactionInitiator);
    public abstract void performInteraction(Entity reactionInitiator);
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/activepoints/reaction/Reaction.kt
package com.gmail.borlandlp.minigamesdtools.activepoints.reaction

import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint
import org.bukkit.entity.Entity

abstract class Reaction {
    lateinit var activePoint: ActivePoint

    abstract fun performDamage(reactionInitiator: Entity, damage: Double)
    abstract fun performIntersection(reactionInitiator: Entity)
    abstract fun performInteraction(reactionInitiator: Entity)
}